src/test/java/org/broadinstitute/hellbender/BwaMemIntegrationTest.java

```diff
@@ -1,34 +1,162 @@
 package org.broadinstitute.hellbender;
 
+import htsjdk.samtools.SAMFlag;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import htsjdk.samtools.reference.FastaSequenceIndex;
+import htsjdk.samtools.reference.FastaSequenceIndexCreator;
+import htsjdk.samtools.reference.IndexedFastaSequenceFile;
+import org.broadinstitute.hellbender.utils.RandomDNA;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.bwa.BwaMemAligner;
+import org.broadinstitute.hellbender.utils.bwa.BwaMemAlignment;
 import org.broadinstitute.hellbender.utils.bwa.BwaMemIndex;
 import org.broadinstitute.hellbender.utils.test.BaseTest;
-import org.testng.annotations.*;
+import org.broadinstitute.hellbender.utils.test.ReadTestUtils;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
 
-public class BwaMemIntegrationTest extends BaseTest {
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+
+public final class BwaMemIntegrationTest extends BaseTest {
 
     private BwaMemIndex index;
+    private File fastaFile;
+    private File imageFile;
+    private FastaSequenceIndex fastaIndex;
+
+    private static final SAMSequenceDictionary TEST_DICTIONARY = new SAMSequenceDictionary(
+            Arrays.asList(new SAMSequenceRecord("chr1", 10_000),
+                          new SAMSequenceRecord("chr2", 10_000))
+    );
 
     @BeforeClass
-    public void loadIndex() {
-        final String imageFile = createTempFile(b37_reference_20_21, ".img").toString();
-        BwaMemIndex.createIndexImageFromFastaFile(b37_reference_20_21, imageFile);
-        index = new BwaMemIndex(imageFile);
+    public void loadIndex() throws IOException {
+        final RandomDNA rdnDNA = new RandomDNA(103);
+        try {
+            fastaFile = rdnDNA.nextFasta(TEST_DICTIONARY, 60);
+            fastaFile.deleteOnExit();
+            imageFile = createTempFile(fastaFile.getAbsolutePath(), ".img");
+            fastaIndex = FastaSequenceIndexCreator.buildFromFasta(fastaFile.toPath());
+            BwaMemIndex.createIndexImageFromFastaFile(fastaFile.getAbsolutePath(), imageFile.getAbsolutePath());
+            index = new BwaMemIndex(imageFile.getAbsolutePath());
+        } catch (final Throwable th) {
```

Comment:
In general, I think it's a bad idea to catch Throwable.  You can Google it to see if you agree.
You're working very hard to delete two files that are already marked for deleteOnExit.
But I'm not that concerned because this is test code.

Reply:
Changed it to RuntimeException.